src/Stratis.Bitcoin.Features.Consensus/Rules/ConsensusRulesExtension.cs

```diff
@@ -24,14 +25,11 @@ public static class ConsensusRulesExtension
         /// <typeparam name="T">The type of rule to find.</typeparam>
         /// <param name="rules">The rules to look in.</param>
         /// <returns>The rule or <c>null</c> if not found in the list.</returns>
-        public static T FindRule<T>(this IEnumerable<ConsensusRule> rules) where T : ConsensusRule
+        public static T FindRule<T>(this IEnumerable<IConsensusRule> rules) where T : ConsensusRule
         {
             var rule = rules.TryFindRule<T>();
-
             if (rule == null)
-            {
                 throw new Exception(); // todo:
```

Comment:
Can you add some message to this exception?

Reply:
Done :)